sound/soc/sof/ipc.c

```diff
@@ -398,17 +398,30 @@ static void ipc_msgs_rx(struct work_struct *work)
 	case SOF_IPC_FW_READY:
 		/* check for FW boot completion */
 		if (!sdev->boot_complete) {
-			if (sof_ops(sdev)->fw_ready)
+			if (sof_ops(sdev)->fw_ready) {
 				err = sof_ops(sdev)->fw_ready(sdev, cmd);
+			} else {
+				err = -ENOTSUPP;
+				dev_err(sdev->dev,
+					"error: fw_ready op not defined\n");
+			}
 			if (err < 0) {
-				dev_err(sdev->dev, "error: DSP firmware boot timeout %d\n",
+				/*
+				 * this indicates either an ABI mismatch or
+				 * that the ABI cannot be verified due to
+				 * missing fw_ready op
+				 */
+				dev_err(sdev->dev,
+					"error: fw ready failed with error %d\n",
 					err);
```

Comment:
You improved error reporting consistency, which is good. But in some cases, I think, the effect can be a bit weird... For instance, if there's no .fw_ready() method defined, you'll first print "op not defined," then "fw ready failed with error" and then below you'll add to that "ABI match failed"... Or if .fw_ready() is defined but did return an error, you'll print two error messages for that. I think 1 error should produce 1 error message. I think it would be enough to decide for 1 of the two approaches: either print errors in this function, which allows for finer granularity, or below, when the waiter is woken up, but maybe not in both...

Reply:
@lyakh ack. will change to only having the error printed in one place.